libminifi/test/TestBase.cpp

```diff
@@ -61,78 +62,51 @@ TestPlan::~TestPlan() {
   for (auto& processor : configured_processors_) {
     processor->setScheduledState(core::ScheduledState::STOPPED);
   }
+  for (auto& connection : relationships_) {
```

Comment:
There was an issue where we could not get processors to stop due to them being cross-referenced by their corresponding connections, holding cross-ownerships. This prevented them to be properly freed in the testcases using `TestBase`.